includes/wizards/class-network-wizard.php

```diff
@@ -55,93 +61,203 @@ public function __construct() {
+			return $screen_slug;
 		}
+		$screen_slug = '';
 
-		// Check for admin post type screen: Listings page and classic editor (add new + edit), but not block editor.
-		if ( ! empty( $current_screen->post_type )
-			&& ! empty( $this->admin_screens['post_type'][ $current_screen->post_type ] )
-			&& false === $current_screen->is_block_editor ) {
-			$this->slug = $current_screen->post_type;
-			$this->screen_type = 'post_type';
-			return $this->enable_display();
+		$sanitized_action    = sanitize_text_field( $_GET['action'] ?? '' ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$sanitized_page      = sanitize_text_field( $_GET['page'] ?? '' ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$sanitized_post_type = sanitize_text_field( $_GET['post_type'] ?? '' ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$sanitized_post_id   = sanitize_text_field( $_GET['post'] ?? '' ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		
+		if ( 'admin.php' === $pagenow && isset( $this->admin_screens[ $sanitized_page ] ) ) {
+			// admin page screen: admin.php?page={page} .
+			$screen_slug = $sanitized_page;
+		} elseif ( 'edit.php' === $pagenow && isset( $this->admin_screens[ $sanitized_post_type ] ) ) {
+			// post type list screen: edit.php?post_type={post_type} .
+			$screen_slug = $sanitized_post_type;
+		} elseif ( 'post-new.php' === $pagenow && 'newspack_hub_nodes' === $sanitized_post_type ) {
+			// add new node screen: post-new.php?post_type=newspack_hub_nodes .
+			// note: assumes non-block editor, otherwise we need to not set this.
+			$screen_slug = $sanitized_post_type;
+		} elseif ( 'post.php' === $pagenow && 'edit' === $sanitized_action && 'newspack_hub_nodes' === get_post_type( $sanitized_post_id ) ) {
+			// edit node screen: post.php?post={ID}&action=edit
+			// note: assumes non-block editor, otherwise we need to not set this.
+			$screen_slug = 'newspack_hub_nodes';
 		}
+
+		return $screen_slug;
 	}
 
 	/**
-	 * Enable display of the Wizard Header with body CSS.
+	 * Wrapper for Network Plugin's is_node/is_hub functions. Also called by Newspack_Dashboard.
 	 */
-	protected function enable_display() {
-		// Add CSS to body.
-		add_filter( 'admin_body_class', [ $this, 'add_body_class' ] );
+	public static function get_site_role() {
+
+		static $site_role;
+		if ( isset( $site_role ) ) {
+			return $site_role;
+		}
+
+		$is_node_function = [ Newspack_Network_Site_Role::class, 'is_node' ];
+		$is_hub_function  = [ Newspack_Network_Site_Role::class, 'is_hub' ];
 
-		// Display Wizard Admin Header.
-		$this->admin_header_init( [ 'title' => $this->get_name() ] );
+		if ( is_callable( $is_node_function ) && call_user_func( $is_node_function ) ) {
+			$site_role = 'node';
+		} elseif ( is_callable( $is_hub_function ) && call_user_func( $is_hub_function ) ) {
+			$site_role = 'hub';
+		} else {
+			$site_role = '';
+		}
+
+		return $site_role;
```

Comment:
This can be simplified.
```suggestion
		static $site_role;
		if ( isset( $site_role ) ) {
			return $site_role;
		}
		$site_role = Newspack_Network_Site_Role::get();
		return $site_role;
```
Below is an even simpler method, slightly less readable:
```php
public static function get_site_role() {
	static $site_role = null;
	$site_role ??= Newspack_Network_Site_Role::get();
	return $site_role;
}
```

Reply:
Thanks for sharing `??=` that's very cool.  I do like the simplified solutions but I have two concerns.  The first concern about not using `is_callable` I'll share in the main comment thread here in a minute.  The second concern is the `Newspack_Network_Site_Role::get()` function could return `false` if the option isn't set yet.  So we'd need to make sure `false` gets changed to `''`.  I went with an in-between solution that reduces the `is_callable`'s down to just one, but still tests that the function exists and changes false to blank.  Please see: https://github.com/Automattic/newspack-plugin/pull/3479/commits/703a8d7e3d2d1c19164765de14d4fb58ba045619